server/src/apub/fetcher.rs

```diff
@@ -109,50 +58,114 @@ pub fn search_by_apub_id(query: &str, conn: &PgConnection) -> Result<SearchRespo
   };
   match fetch_remote_object::<SearchAcceptedObjects>(&query_url)? {
     SearchAcceptedObjects::Person(p) => {
-      let u = upsert_user(&UserForm::from_person(&p)?, conn)?;
-      response.users = vec![UserView::read(conn, u.id)?];
+      let user_uri = p.base.base.object_props.get_id().unwrap().to_string();
+      let user = get_or_fetch_and_upsert_remote_user(&user_uri, &conn)?;
+      response.users = vec![UserView::read(conn, user.id)?];
     }
     SearchAcceptedObjects::Group(g) => {
-      let c = upsert_community(&CommunityForm::from_group(&g, conn)?, conn)?;
-      fetch_community_outbox(&c, conn)?;
-      response.communities = vec![CommunityView::read(conn, c.id, None)?];
-    }
-    SearchAcceptedObjects::Page(p) => {
```

Comment:
Why did you specifically remove searching for post ID? I dont see any reason for that.

Reply:
See comments above, we can add it back in later, if we ever want to support history, but for now, lets focus on only activities posted to our inboxes, and the only fetching we're doing is for actors.

Reply:
I'm okay with removing the functionality to read posts from a community outbox, because that is complicated for various reasons. 

But this is just fetching a post by its ID, and has nothing at all to do with the outbox. I dont see how its "not really the activitypub way". In fact, every activitypub implementation that I know supports this. So please put it back, I wont merge a PR that removes a perfectly working feature for no reason at all.

Reply:
Okay, I've added it back in.

Reply:
Thanks!